app/forms/user_form.rb

```diff
@@ -2,11 +2,65 @@
 
 ##
 # Form object for creating or updating a User
+require 'digest/sha1'
+
 class UserForm
-  include AuthenticationForm
-  include Auditor
+  include ActiveModel::Model
+  include FormObject
+
+  set_form_variables :user_code, current_user: :find_current_user
 
-  set_attributes :login, :swipe_card_id, :barcode, :type, :status, :team_id
+  validate :check_user
 
   delegate :becomes, to: :user
+
+  def submit(params)
+    @params = params
+    assign_params
+    @current_user = find_current_user
+
+    case @params[:commit]
```

Comment:
I think there is a different way to do this based on the object that is passed because I am sure that if it is an existing user it will have an id but not crucial.